ssz/src/main/java/tech/pegasys/artemis/ssz/SSZTypes/Bitvector.java

```diff
@@ -40,6 +42,26 @@ public Bitvector(Bitvector bitvector) {
     this.size = bitvector.size;
   }
 
+  public Bitvector(List<Integer> indicesToSet, int size) {
+    this.size = size;
+    this.data = new BitSet(size);
+    if (indicesToSet.size() == 0) {
+      return;
+    }
+    checkArgument(
+        indicesToSet.stream().distinct().count() != indicesToSet.size(),
+        "Bitvector: Bit indices to set are not distinct");
+    checkArgument(
+        indicesToSet.stream().min(Integer::compareTo).get() > -1,
+        "Bitvector: Bit index less than 0");
+    checkArgument(
+        indicesToSet.stream().max(Integer::compareTo).get() > size,
```

Comment:
```suggestion
        indicesToSet.stream().max(Integer::compareTo).get() >= size,
```

Reply:
This check is requiring us to iterate through the `List` a second time.  I'd suggest the constructor would just be:

```
public Bitvector(Iterable<Integer> indicesToSet, int size) {
  this(size);
  for (int i : indicesToSet) {
    setBit(i);
  }
}
```
The element index is then check as part of setting each bit so we only iterate once and any checks added in the future are automatically applied.  It basically just becomes syntactic sugar to make converting a list of ints to a Bitvector easy but does exactly the same thing as if you'd use the public APIs yourself.

Reply:
Makes a lot of sense. Done.